Add `home_connect.set_program_and_options` (#37468)
home-assistant/home-assistant.github.io: source/_integrations/home_connect.markdown

## Actions

The Home Connect integration makes various actions available.
Available actions: `set_program_and_options`, `set_option_active`, `set_option_selected`, `pause_program`, `resume_program`, `select_program`, `start_program` and `change_setting`

### Action `home_connect.set_program_and_options`

> Starts or selects a program, or if `program` attribute is not set, sets the options for the active or the selected program.

Reviewer: ```suggestion
Starts or selects a program. If the `program` attribute is not set, this action sets the options for the active or the selected program.
```

| Data attribute    | Optional | Description                                      |
|---------------------------|----------|--------------------------------------------------|
| `device_id` | no | ID of the device. |
| `affects_to` | no | Selects if the program affected by the action should be the active or the selected program. |
| `program` | yes | Program to select. If set, it will start or select a program depending on `affects_to`. |

> | `consumer_products_cleaning_robot_option_reference_map_id` | yes | Defines the used reference map. |

Reviewer: ```suggestion
| `consumer_products_cleaning_robot_option_reference_map_id` | yes | Defines which reference map is to be used. |
```

| `consumer_products_cleaning_robot_option_cleaning_mode` | yes | Defines the favoured cleaning mode. |

> | `consumer_products_coffee_maker_option_bean_amount` | yes | Describes the bean amount of a coffee machine program. |

Reviewer: ```suggestion
| `consumer_products_coffee_maker_option_bean_amount` | yes | Describes the amount of coffee beans used in a coffee machine program. |
```

> | `consumer_products_coffee_maker_option_fill_quantity` | yes | Describes the fill quantity (in ml) of a coffee machine program. |

Reviewer: ```suggestion
| `consumer_products_coffee_maker_option_fill_quantity` | yes | Describes the amount of water (in ml) used in a coffee machine program. |
```

> | `consumer_products_coffee_maker_option_coffee_temperature` | yes | Describes the coffee temperature of a coffee machine program. |

Reviewer: ```suggestion
| `consumer_products_coffee_maker_option_coffee_temperature` | yes | Describes the coffee temperature used in a coffee machine program. |
```

| `consumer_products_coffee_maker_option_bean_container` | yes | Defines the preferred bean container. |
| `consumer_products_coffee_maker_option_flow_rate` | yes | Defines the water-coffee contact time. The duration extends to coffee intensity. |
| `consumer_products_coffee_maker_option_multiple_beverages` | yes | Defines if double dispensing is enabled. |

> | `consumer_products_coffee_maker_option_coffee_milk_ratio` | yes | Defines the milk amount. |

Reviewer: ```suggestion
| `consumer_products_coffee_maker_option_coffee_milk_ratio` | yes | Defines the amount of milk. |
```

| `consumer_products_coffee_maker_option_hot_water_temperature` | yes | Defines the temperature suitable for the type of tea. |

> | `b_s_h_common_option_start_in_relative` | yes | Defines in how many time the program should start. |

Reviewer: I'm not sure I understand this. by "How many time" do you mean how often (as in run twice, to fill 2 cups)? Or at which time (start at 6:00)? Or do you mean a time delta, as in "in 30 minutes from now"?

Author: https://api-docs.home-connect.com/programs-and-options/#dishwasher_start-in-relative-option

Is a time delta

> | `b_s_h_common_option_start_in_relative` | yes | Defines in how many time the program should start. |

Reviewer: ```suggestion
| `b_s_h_common_option_start_in_relative` | yes | Defines when the program should start. For example: in 2 h 30 min.|
```
Ah ok. So maybe we could add an example like this?

Author: The thing is that it uses an input number that represents the relative time in seconds, so adding an example that says "2h and 30 min" might be confusing

Reviewer: then I think it should definitely be mentioned. It's not what I would expect. My devices go in 30 minute steps.

"Defines when the program should start, in seconds from now. For example: a value of 9000 means in 2 h 30 min.|"

| `dishcare_dishwasher_option_intensiv_zone` | yes | Defines if the cleaning is done with higher spray pressure on the lower basket for very dirty pots and pans. |

> | `dishcare_dishwasher_option_brilliance_dry` | yes | Defines if the program sequence is optimized with special drying cycle ensures more shine on glasses and plastic items. |

Reviewer: ```suggestion
| `dishcare_dishwasher_option_brilliance_dry` | yes | Defines if the program sequence is optimized with a special drying cycle to ensure more shine on glasses and plastic items. |
```

> | `dishcare_dishwasher_option_vario_speed_plus` | yes | Defines if the program run time is reduced by up to 66% with the usual optimum cleaning and drying. |

Reviewer: do you mean reduced to 66 percent of the original runtime? Or reduced by 66% (so that the new runtime equals 44% of the original runtime)?

Author: Idk 😅
Copypasted from docs: https://api-docs.home-connect.com/programs-and-options/#dishwasher_variospeed-plus-option

> | `dishcare_dishwasher_option_vario_speed_plus` | yes | Defines if the program run time is reduced by up to 66% with the usual optimum cleaning and drying. |

Reviewer: ```suggestion
| `dishcare_dishwasher_option_vario_speed_plus` | yes | Defines if the program duration is shortened dynamically (up to 66% less run time) with the usual optimum cleaning and drying. |
```
Found it! How about we say it like this?
![image](https://github.com/user-attachments/assets/ae2b7926-f2e0-493e-bbe7-74b7a6e0b7dc)

| `dishcare_dishwasher_option_silence_on_demand` | yes | Defines if the extra silent mode is activated for a selected period of time. |

> | `dishcare_dishwasher_option_half_load` | yes | Defines if economical cleaning is enabled for smaller loads which reduces energy and water consumption and also saves time. The utensils can be placed in the upper and lower baskets. |

Reviewer: ```suggestion
| `dishcare_dishwasher_option_half_load` | yes | Defines if economical cleaning is enabled for smaller loads. This reduces energy and water consumption and also saves time. The utensils can be placed in the upper and lower baskets. |
```

| `dishcare_dishwasher_option_extra_dry` | yes | Defines if improved drying for glasses and plasticware is enabled. |

> | `dishcare_dishwasher_option_hygiene_plus` | yes | Defines if the cleaning is done with increased temperatures which ensures maximum hygienic cleanliness for regular use. |

Reviewer: ```suggestion
| `dishcare_dishwasher_option_hygiene_plus` | yes | Defines if the cleaning is done with increased temperatures. This ensures maximum hygienic cleanliness for regular use. |
```

| `dishcare_dishwasher_option_eco_dry` | yes | Defines if the door is opened automatically for extra energy efficient and effective drying. |
| `dishcare_dishwasher_option_zeolite_dry` | yes | Defines if the program sequence is optimized with special drying cycle ensures improved drying for glasses, plates and plasticware. |

> | `laundry_care_dryer_option_drying_target` | yes | Describes the drying target for a dryer program. |

Reviewer: what is a drying target? are there options, like moist, dry, extra dry?

Author: from https://api-docs.home-connect.com/programs-and-options/#dryer_drying-target-option, the possible options are:

- Iron Dry
- Gentle Dry
- Cupboard Dry
- Cupboard Dry Plus
- Extra Dry

> | `laundry_care_dryer_option_drying_target` | yes | Describes the drying target for a dryer program. |

Reviewer: ```suggestion
| `laundry_care_dryer_option_drying_target` | yes | Describes the drying target for a dryer program. For example: Iron Dry, Cupboard Dry, Extra Dry. |
```
ah ok. Thanks for the extra info.

Author: Should we add examples for every enumeration?

Reviewer: wherever you find useful. I added some here because I wasn't sure what "drying target" meant.

| `cooking_hood_option_venting_level` | yes | Defines the required fan setting. |
| `cooking_hood_option_intensive_level` | yes | Defines the intensive setting. |
| `cooking_oven_option_setpoint_temperature` | yes | Defines the target cavity temperature, which will be hold by the oven. |

> | `b_s_h_common_option_duration` | yes | Defines the run-time of the program. Afterwards the appliance is stopped. |

Reviewer: ```suggestion
| `b_s_h_common_option_duration` | yes | Defines the run-time of the program. Afterwards, the appliance is stopped. |
```

> | `cooking_oven_option_fast_pre_heat` | yes | Defines if the cooking compartment is heated up quickly. Please note that the setpoint temperature has to be equal or higher than 100 °C or 212 °F otherwise the fast pre-heat option is not activated. |

Reviewer: ```suggestion
| `cooking_oven_option_fast_pre_heat` | yes | Defines if the cooking compartment is heated up quickly. Please note that the setpoint temperature has to be equal to or higher than 100 °C or 212 °F. Otherwise, the fast pre-heat option is not activated. |
```
